eidpauth/src/main/java/uk/ac/ucl/eidp/auth/jaas/SingleFactorLoginModule.java

```diff
@@ -83,12 +83,21 @@ public boolean login() throws LoginException {
     PasswordCallback passwordCallback = new PasswordCallback("Password : ", false);
     try {
       callbackHandler.handle(new Callback[] {nameCallback, passwordCallback});
+
+      char[] passwordChar = passwordCallback.getPassword();
+      String password = null;
+      if (null != passwordChar) {
+        password = new String(passwordChar);
+      }
       String username = nameCallback.getName();
-      String password = new String(passwordCallback.getPassword());
       nameCallback.setName("");
       passwordCallback.clearPassword();
 
-      UserE userE = userController.findUser(username, password);
+      UserE userE = null;
+      
+      if (null == userController) {
+        userE = userController.findUser(username, password);
```

Comment:
\[FindBugs\] WARNING: NP: Null pointer dereference of SingleFactorLoginModule\.userController in uk\.ac\.ucl\.eidp\.auth\.jaas\.SingleFactorLoginModule\.login\(\)